mono/mini/debugger-agent.c

```diff
@@ -4967,8 +4967,12 @@ ss_create_init_args (SingleStepReq *ss_req, SingleStepArgs *args)
 				found_sp = mono_find_prev_seq_point_for_native_offset (frame->de.domain, frame->de.method, frame->de.native_offset, &info, &args->sp);
 				if (!found_sp)
 					no_seq_points_found (frame->de.method, frame->de.native_offset);
-				g_assert (found_sp);
-				method = frame->de.method;
+#if !defined(HOST_ANDROID) && !defined(TARGET_ANDROID)					
+				g_assert (found_sp);				
+#else
+				if (found_sp) //it will not find the next sp if it is a "DynamicMethodNameCounter" from Android.Runtime
```

Comment:
This comment isn't very clear.  Do you mean this: https://github.com/xamarin/xamarin-android/blob/0ec01060f6bc16c55fdc2e2dae9c21b34a94e7bb/src/Mono.Android/Android.Runtime/JNINativeWrapper.cs#L52 ?

Reply:
The name of the class of the method that is on stack is DynamicMethodNameCounter and the name of the method is in this case 5. I'm not sure if this is the case...

Reply:
Yea, I think it's that line.  Note that the "owner" parameter is `typeof (DynamicMethodNameCounter)`: https://docs.microsoft.com/en-us/dotnet/api/system.reflection.emit.dynamicmethod.-ctor?view=netframework-4.8#System_Reflection_Emit_DynamicMethod__ctor_System_String_System_Reflection_MethodAttributes_System_Reflection_CallingConventions_System_Type_System_Type___System_Type_System_Boolean_

And then in `reflection_create_dynamic_method` in Mono we use that `owner` parameter to set the `klass` of the method.  And I think the debugger loses track of it.

Reply:
Thanks for the explanation! :)